packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4279,6 +4279,60 @@ class InputDecorationTheme with Diagnosticable {
     );
   }
 
+  /// Returns a copy of this InputDecorationTheme where the non-null fields in
+  /// the given InputDecorationTheme override the corresponding fields in this
+  /// InputDecorationTheme.
+  ///
+  /// In other words, [inputDecorationTheme]'s fields are used to fill in the
+  /// (unspecified) fields of this InputDecorationTheme.
+  InputDecorationTheme merge(InputDecorationTheme? inputDecorationTheme) {
+    if (inputDecorationTheme == null) {
+      return this;
+    }
+    return copyWith(
+      labelStyle: labelStyle ?? inputDecorationTheme.labelStyle,
+      floatingLabelStyle: floatingLabelStyle ?? inputDecorationTheme.floatingLabelStyle,
+      helperStyle: helperStyle ?? inputDecorationTheme.helperStyle,
+      helperMaxLines: helperMaxLines ?? inputDecorationTheme.helperMaxLines,
+      hintStyle: hintStyle ?? inputDecorationTheme.hintStyle,
+      errorStyle: errorStyle ?? inputDecorationTheme.errorStyle,
+      errorMaxLines: errorMaxLines ?? inputDecorationTheme.errorMaxLines,
+      floatingLabelBehavior: floatingLabelBehavior == inputDecorationTheme.floatingLabelBehavior
+        ? floatingLabelBehavior
+        : inputDecorationTheme.floatingLabelBehavior,
+      floatingLabelAlignment: floatingLabelAlignment == inputDecorationTheme.floatingLabelAlignment
+        ? floatingLabelAlignment
+        : inputDecorationTheme.floatingLabelAlignment,
```

Comment:
These aren't nullable. 

I'm not sure if it should not be possible to override these properties in `InputDecorationTheme.merge` or if we can override them by checking if the same value or not. Similar `merge` methods in the framework don't have this challenge.

I did the latter here, only updating the value if it is not the same but I'm open to suggestions on how to handle these non-nullable properties.

Reply:
The approach you've taken makes sense but it's a little subtle. I think it would be a little more obvious to not merge the non-nullable fields. That way the definition of what the merge method does remains straightforward.